src/storage/src/hummock/store/version.rs

```diff
@@ -464,16 +714,31 @@ impl HummockVersionReader {
         // Because SST meta records encoded key range,
         // the filter key needs to be encoded as well.
         assert!(committed_version.is_valid());
-        for level in committed_version.levels(read_options.table_id) {
+        for (table_view_position, level) in committed_version
+            .levels(read_options.table_id)
+            .iter()
+            .enumerate()
+        {
             if level.table_infos.is_empty() {
                 continue;
             }
 
             match level.level_type() {
                 LevelType::Overlapping | LevelType::Unspecified => {
+                    // TODO: Use an elegant way to avoid duplicate code.
+                    let full_idx = (0..level.get_table_infos().len()).collect_vec();
+                    let sst_indices = committed_index
+                        .as_deref()
+                        .and_then(|pruned_version| {
+                            pruned_version
+                                .get_indices(table_view_position, level.get_sub_level_id())
```

Comment:
> The logic here is a bit tricky and prone to bugs. `table_view_position` is obtained via `committed_version.levels(read_options.table_id).iter().enumerate()`, which means the ordering of `PrunedVersion::levels` must be consistent with `committed_version.levels(...)`. The ordering guarantee is currently maintained very implicitly in the implementation of `PrunedVersion` and `PinnedVersion`.
> 
> Can we use a HashMap/BTreeMap to store the mapping from sub level id to sub level table indices in PrunedVersion so that we don't need to rely on the ordering of different data structures to ensure correctness? By doing so, I think we can also simplify `PrunedVersion::apply`.

I have considered about this but I thought that `BTreeMap` is slow.